### PR TITLE
Upgrade test contracts to 0.5.0 and fix all resulting problems

### PR DESCRIPTION
**Pull request by haltman-at**
[Note: This branch is kind of a mess; it might not end up getting merged in as such.]

This branch upgrades all the test contracts to 0.5.0, and attempts to fix all the resulting problems.  I've already fixed one such problem, namely the lack of support for `STATICCALL`, in PR #1446.  But there's more.

One issue that's come up is that a number of my tests are quite shaky.  This isn't an 0.5.0 problem, but I in my investigations I think I've figured out what's causing it anyway -- namely, I was relying on `sourceId` to be consistent between runs, when it's not.  That's an easy fix.

But the real problems are the 0.5.0-specific problems, which are very consistent.  The failing tests (now that I've fixed the storage decoding tests, where the problem was `STATICCALL`) are:
1. The memory decoding tests
2. The variable IDs tests
3. The two mapping tests where the domain is numeric

Note that these errors exist also, in near-identical form, on the `truffle-decoder` branch; I made a separate branch for testing that, namely `decoder-0.5.0-tests`, which see.

The memory decoding tests, as best I can tell, do seem to be due to a decoding problem.  However, I'm not certain of this.

The variable ID tests seem to be due to some difference in how variable declarations are handled, except for one of them which also seems to be due to a decoding problem like above.  Again, all this is uncertain.

The two mapping tests... I'd say also a decoding problem, probably, but a different one.  However I've made no real exploration of this at all.

Basically, I don't know what's going on, but that's my current progress.  Will update.

### CODE REVIEW

**Comment by haltman-at** [comment]
Note: The intermittent failure should now be addressed due to PR #1466.

File: packages/truffle-debugger/test/helpers.js [hunk 1]
```diff
@@ -68,17 +62,21 @@ export function getAccounts(provider) {
...
-    ) {
-      if (err) return reject(err);
-      result.resolver = new Resolver(result);
-      result.artifactor = new Artifactor(result.contracts_build_directory);
-      result.networks = {};
+    Box.sandbox(
+      {
+        unsafeCleanup: true,
+        setGracefulCleanup: true,
+        name: "default#web3-one"
```
Comment (gnidan) [inline packages/truffle-debugger/test/helpers.js]: This change is necessary for Migrations.sol

File: packages/truffle-config/index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-unused-vars */
```
Comment (gnidan) [inline packages/truffle-config/index.js]: We should probably get rid of this

File: packages/truffle-debugger/test/ast.js [hunk 3]
```diff
@@ -13,10 +12,10 @@ import ast from "lib/ast/selectors";
 import solidity from "lib/solidity/selectors";
 import trace from "lib/trace/selectors";
 
-import { getRange, findRange, rangeNodes } from "lib/ast/map";
+import { getRange } from "lib/ast/map";
 
 const __VARIABLES = `
-pragma solidity ^0.4.18;
+pragma solidity ~0.5;
```
Comment (gnidan) [inline packages/truffle-debugger/test/ast.js]: These of course are necessary

File: packages/truffle-debugger/test/context.js [hunk 4]
```diff
@@ -11,7 +11,7 @@ import Debugger from "lib/debugger";
 import sessionSelector from "lib/session/selectors";
 
 const __OUTER = `
-pragma solidity ^0.4.18;
+pragma solidity ~0.5;
```
Comment (gnidan) [inline packages/truffle-debugger/test/context.js]: :+1:

File: packages/truffle-debugger/test/context.js [hunk 5]
```diff
@@ -33,7 +33,7 @@ contract OuterContract {
 `;
 
 const __INNER = `
-pragma solidity ^0.4.18;
+pragma solidity ~0.5;
```
Comment (gnidan) [inline packages/truffle-debugger/test/context.js]: :+1:

File: packages/truffle-debugger/test/data/decode/decoding.js [hunk 6]
```diff
@@ -5,136 +5,155 @@ import faker from "faker";
...
    */
   describeDecoding(
-    "Storage Variables", commonFixtures, evm.current.state.storage,
+    "Storage Variables",
+    commonFixtures,
+    evm.current.state.storage,
 
     (contractName, fixtures) => {
-      return `pragma solidity ^0.4.23;
+      return `pragma solidity ~0.5;
```
Comment (gnidan) [inline packages/truffle-debugger/test/data/decode/decoding.js]: :+1:

File: packages/truffle-debugger/test/data/decode/decoding.js [hunk 7]
```diff
@@ -5,136 +5,155 @@ import faker from "faker";
...
 
   describeDecoding(
-    "Mapping Variables", mappingFixtures, evm.current.state.storage,
+    "Mapping Variables",
+    mappingFixtures,
+    evm.current.state.storage,
 
     (contractName, fixtures) => {
-      return `pragma solidity ^0.4.24;
+      return `pragma solidity ~0.5;
```
Comment (gnidan) [inline packages/truffle-debugger/test/data/decode/decoding.js]: :+1:

File: packages/truffle-debugger/test/data/decode/helpers.js [hunk 8]
```diff
@@ -67,20 +65,20 @@ async function prepareDebugger(testName, sources) {
...
+  );
   let sourceId = matchingSources[0].id;
 
   let breakpoint = {
     sourceId,
     line: lastStatementLine(source)
   };
 
-  session.addBreakpoint(breakpoint,true);
+  session.addBreakpoint(breakpoint);
```
Comment (gnidan) [inline packages/truffle-debugger/test/data/decode/helpers.js]: Sure if you wanna include

File: packages/truffle-debugger/test/data/ids.js [hunk 9]
```diff
@@ -13,7 +13,7 @@ import trace from "lib/trace/selectors";
 import solidity from "lib/solidity/selectors";
```
Comment (gnidan) [inline packages/truffle-debugger/test/data/ids.js]: All the things in this file

File: packages/truffle-debugger/test/endstate.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
 import debugModule from "debug";
```
Comment (gnidan) [inline packages/truffle-debugger/test/endstate.js]: all the things here

File: packages/truffle-debugger/test/evm.js [hunk 11]
```diff
@@ -1,10 +1,9 @@
 import debugModule from "debug";
```
Comment (gnidan) [inline packages/truffle-debugger/test/evm.js]: yep these too

File: packages/truffle-debugger/test/helpers.js [hunk 12]
```diff
@@ -23,12 +23,6 @@ export async function prepareContracts(provider, sources = {}, migrations) {
   };
   config.network = "debugger";
 
-  config.compilers = {
-    solc: {
-      version: "0.4.25"
-    }
-  };
-
```
Comment (gnidan) [inline packages/truffle-debugger/test/helpers.js]: Gotta put this back in and turn the version up, apparently!

File: packages/truffle-debugger/test/precompiles.js [hunk 13]
```diff
@@ -1,10 +1,9 @@
 import debugModule from "debug";
```
Comment (gnidan) [inline packages/truffle-debugger/test/precompiles.js]: All this

File: packages/truffle-debugger/test/reset.js [hunk 14]
```diff
@@ -12,14 +12,14 @@ import data from "lib/data/selectors";
 import solidity from "lib/solidity/selectors";
```
Comment (gnidan) [inline packages/truffle-debugger/test/reset.js]: yep

File: packages/truffle-debugger/test/solidity.js [hunk 15]
```diff
@@ -12,7 +12,7 @@ import solidity from "lib/solidity/selectors";
 import trace from "lib/trace/selectors";
```
Comment (gnidan) [inline packages/truffle-debugger/test/solidity.js]: yep

**Comment by haltman-at** [comment]
OK, the optimizer has now been turned off for the tests (thanks @gnidan), solving most of the problems.

In addition, I've passed the `files` argument to `prepareDebugger` in tests that were previously missing it, which should (independently of the `0.5.0` update) fix their intermittent failures.

A word of warning, the compiler version is currently set to `0.5.0` rather than `^0.5.0`; this is because the latter doesn't work on my machine.  We'll have to figure out some way of getting the latter to work, because we don't want to have to constantly update the version every time there's a new version of `solc`.
